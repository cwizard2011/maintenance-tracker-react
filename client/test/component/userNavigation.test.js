import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import DefaultUserNavigation, { UserNavigation } from '../../src/components/UserNavigation';

const mockStore = configureMockStore([thunk]);

const props = {
  user: {
    email: 'juliet',
    exp: 1540907850,
    iat: 1540886250,
    id: '48a698a0-1641-5aca-bc1b-de9b1a482ee1',
    user_role: 'user',
    username: 'juliet'
  },
  auth: true,
  logout: jest.fn()
};
const initialState = {
  authReducer: {
    user: {
      email: 'juliet',
      exp: 1540907850,
      iat: 1540886250,
      id: '48a698a0-1641-5aca-bc1b-de9b1a482ee1',
      user_role: 'user',
      username: 'juliet'
    },
    isAuthenticated: true
  }
};
const store = mockStore(initialState);

describe('User Navigation', () => {
  it('render user navigation correctly', () => {
    const wrapper = shallow(<UserNavigation {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render connected component', () => {
    const wrapper = shallow(<DefaultUserNavigation store={store} {...props} />);
    expect(wrapper.state()).toEqual({});
  });
  it('should logout a user when logout is clicked', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    const wrapper = shallow(<UserNavigation {...props} />);
    const instance = wrapper.instance();
    instance.handleLogout(event);
    expect(props.logout).toHaveBeenCalled();
  });
});
