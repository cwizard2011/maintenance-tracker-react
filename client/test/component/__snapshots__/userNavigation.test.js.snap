// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Navigation render user navigation correctly 1`] = `
<nav
  className="nav"
>
  <div
    className="navWide"
  >
    <div
      className="wideDiv"
    >
      <Link
        replace={false}
        title="Home page"
        to="/"
      >
        <i
          className="fa icon-font"
        >
          
        </i>
      </Link>
      <a
        onClick={[Function]}
        role="presentation"
        title="Login Users"
        value="Show Login"
      >
        Logout
      </a>
      <a
        className="profile-name"
        role="presentation"
        title="Welcome Users"
        value="Display User"
      >
        Welcome juliet
      </a>
    </div>
  </div>
  <div
    className="navNarrow"
  >
    <Link
      replace={false}
      title="Home page"
      to="/"
    >
      <i
        className="fa icon-font"
      >
        
      </i>
    </Link>
    <i
      className="fa fa-bars fa-2x icon-padding"
      onClick={[Function]}
      role="presentation"
    />
    <div
      className="narrowLinks"
    >
      <a
        onClick={[Function]}
        role="presentation"
        title="Login Users"
        value="Show Login"
      >
        Logout
      </a>
      <a
        className="profile-name"
        role="presentation"
        title="Welcome Users"
        value="Display User"
      >
        Welcome juliet
      </a>
    </div>
  </div>
</nav>
`;
